Source/WebCore/dom/Document.cpp

```diff
@@ -1409,8 +1409,13 @@ template<typename NameType>
 static ExceptionOr<Ref<Element>> createHTMLElementWithNameValidation(TreeScope& treeScope, Document& document, const NameType& name)
 {
     RefPtr element = HTMLElementFactory::createKnownElement(name, document);
-    if (LIKELY(element))
+    if (LIKELY(element)) {
+#if ENABLE(CONTENT_EXTENSIONS)
+        if (RefPtr iframe = dynamicDowncast<HTMLIFrameElement>(element.get()))
```

Comment:
This is a hot code path, seems bad to do a branch for all element types when this is only for iframes (which are not constructed as frequently as other nodes). Why isn't this in the HTMLIFrameElement constructor? This is what `HTMLElementFactory::createKnownElement()` calls above for iframes anyway.